UnitystationLauncher/ViewModels/SecurityPanelViewModel.cs

```diff
@@ -0,0 +1,538 @@
+
+        // Do something with the common files
+        foreach (FileInfo file in commonFiles)
+        {
+            var sourceFile = sourceDirInfo.GetFiles().FirstOrDefault(x => x.Name == file.Name);
+            if (sourceFile == null)
+            {
+                throw new FileNotFoundException("Common files don't match somehow?!");
+            }
+
+            var targetFile = targetDirInfo.GetFiles().FirstOrDefault(x => x.Name == file.Name);
+            if (targetFile == null)
+            {
+                throw new FileNotFoundException("Common files don't match somehow?!");
+            }
+
+            File.Copy(sourceFile.FullName,
+                targetFile.FullName, true);
+            Console.WriteLine($"Common file: {file.FullName}");
+        }
+
+        Console.WriteLine("Common files listed.");
+    }
+
+
+    class FileInfoComparer : IEqualityComparer<FileInfo>
+    {
+        public bool Equals(FileInfo? x, FileInfo? y)
+        {
+            if (x == null || y == null) return false;
+            return x.Name.Equals(y.Name, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public int GetHashCode(FileInfo obj)
+        {
+            return obj.Name.GetHashCode();
+        }
+    }
+
+    public void OnScan()
+    {
+        Action<string> info = new Action<string>((string log) => { });
+        Action<string> errors = new Action<string>((string log) => { });
+
+        // Create a DirectoryInfo object for the directory
+        DirectoryInfo directory = new DirectoryInfo(System.AppDomain.CurrentDomain.BaseDirectory);
+
+        DirectoryInfo stagingDirectory = directory.CreateSubdirectory("staging"); //TODO temp
+        DirectoryInfo processingDirectory = directory.CreateSubdirectory("processing");
+        DirectoryInfo? dataPath = null;
+
+        try
+        {
+            DeleteContentsOfDirectory(processingDirectory);
+            info.Invoke("Copying files");
+            CopyFilesRecursively(stagingDirectory.ToString(), processingDirectory.ToString());
+
+            info.Invoke("Cleaning out Dlls and Executables");
+            DeleteFilesWithExtension(processingDirectory.ToString(), "exe");
+            DeleteFilesWithExtension(processingDirectory.ToString(), "dll");
+
+            // Get all files in the directory
+            var directories = processingDirectory.GetDirectories();
+
+
+            // Loop through each file
+            foreach (var directorie in directories)
+            {
+                if (directorie.Name.Contains("_Data"))
+                {
+                    if (dataPath != null)
+                    {
+                        errors.Invoke("oh God 2 Datapaths Exiting!!!");
+                        return;
+                    }
+
+                    dataPath = directorie;
+                }
+            }
+
+            if (dataPath == null)
+            {
+                errors.Invoke("oh God NO Datapath Exiting!!!");
```

Comment:
These log messages should probably go through Serilog so they are handled the same way as the rest of the launcher.

```csharp
Log.Error("oh God NO Datapath Exiting!!!");
```

Reply:
how do I get the incidence of Serilog ?

Reply:
*instance

Reply:
Its static, so just doing that should work